server/common/utils/index.test.js

```diff
@@ -0,0 +1,391 @@
+
+    const result = isUserFavourite(favIds, userId);
+    expect(result).toBe(false);
+  });
+});
+
+describe('function responseWithList', () => {
+  const list = listMock[0];
+  const userId = ObjectId();
+
+  list.cohortId = ObjectId();
+
+  const result = responseWithList(list, userId);
+
+  it('should return object with desired properties', () => {
+    expectedListProperties.map(property =>
+      expect(result).toHaveProperty(property)
+    );
+  });
+
+  it('should return object without favIds property', () => {
+    const notExpected = 'favIds';
+
+    expect(result).not.toHaveProperty(notExpected);
+  });
+
+  it('should return object with desired properties for cohort list', () => {
+    expectedCohortListProperties.map(property =>
+      expect(result).toHaveProperty(property)
+    );
+  });
+});
+
+describe('function responseWithLists', () => {
+  const userId = ObjectId();
+  const lists = responseWithLists(listMock, userId);
+
+  it('expect to have only desired meta-data properties without sensitive data', () => {
+    lists.map(list =>
+      expectedListMetaDataProperties.map(property =>
+        expect(list).toHaveProperty(property)
+      )
+    );
+  });
+
+  const notExpected = ['favIds', 'items'];
+
+  it('returned object should not have items and favIds properties', () => {
+    lists.map(list =>
+      notExpected.map(property => expect(list).not.toHaveProperty(property))
+    );
+  });
+});
+
+describe('function checkIfCurrentUserVoted', () => {
+  it('should be true', () => {
+    const userId = '123';
+    const item = {
+      voterIds: ['123', '345', '678']
+    };
+    const result = checkIfCurrentUserVoted(item, userId);
+
+    expect(result).toBe(true);
+  });
+
+  it('should return true', () => {
+    const userId = ObjectId();
+    const item = { voterIds: ['123', '345', '678'] };
+    item.voterIds.push(userId);
+
+    const result = checkIfCurrentUserVoted(item, userId);
+
+    expect(result).toBe(true);
+  });
+
+  it('should return false', () => {
+    const userId = ObjectId();
+    const item = { voterIds: ['123', '345', '678'] };
+    const result = checkIfCurrentUserVoted(item, userId);
+
+    expect(result).toBe(false);
+  });
+});
+
+describe('funtion responseWithItems ', () => {
+  const userId = ObjectId();
+  const result = responseWithItems(userId, listMock[0]);
+
+  it('should return items with desired properties', () => {
+    expectedItemProperties.map(property =>
+      result.map(item => expect(item).toHaveProperty(property))
+    );
+  });
+
+  const notExpected = 'voterIds';
+  it('should return items without voterIds property', () => {
+    result.map(item => expect(item).not.toHaveProperty(notExpected));
+  });
+});
+
+describe('function responseWithItem', () => {
+  const item = singleItemMock[0];
+  const userId = ObjectId();
+  const result = responseWithItem(item, userId);
+
+  it('should return single item with desired properties', () => {
+    expectedItemProperties.map(property =>
```

Comment:
how does this check 'should return single item`?

Reply:
it's named badly.
Should be `should return the item with desired properties`